lua/telescope/actions/init.lua

```diff
@@ -143,6 +143,87 @@ function actions.toggle_selection(prompt_bufnr)
   current_picker:toggle_selection(current_picker:get_selection_row())
 end
 
+--- Apply `f` to entries of current picker and returns list of mapped entries
+--- `f` takes (entry, index, row) as viable arguments in order
+---@param prompt_bufnr number: The prompt bufnr
+---@param f function: function to apply on entries of picker
+---@return table: result from mapped entries
+function actions.map_entries(prompt_bufnr, f)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local results = {}
+  local index = 1
+  local row, result
+    -- indices are 1-indexed, rows are 0-indexed
+  for entry in current_picker.manager:iter() do
+    row = current_picker:get_row(index)
+    -- assign result to variable as function w/o return value
+    -- results in error upon table.insert as it's not technically nil but 'empty'
+    result = f(entry, index, row)
+    table.insert(results, result)
+    index = index + 1
```

Comment:
We already have the index you can then also just do `results[index] = result` which is faster than `table.insert`